crates/steel-core/src/compiler/passes/mod.rs

```diff
@@ -131,11 +131,7 @@ pub trait Folder {
     }
 
     #[inline]
-    fn visit_vector(&mut self, mut v: Vector) -> ExprKind {
-        let args: Vec<_> = v.args.into_iter().map(|exp| self.visit(exp)).collect();
```

Comment:
I've revisited a few of the visitor traits, trying to ascertain whether it makes sense or not to descend into the vector contents (given that they're self-quoting).

I don't have a _lot_ of confidence in the end result, given that different visitors are used at different "expansion phases", and my understanding of those is not great.

Reply:
Given that they're self quoting, I think it is correct to not descend into the vector contents